android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/shapes/PointUtil.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.barteksc.pdfviewer.annotation.core.shapes
+
+import android.graphics.PointF
+import com.github.barteksc.pdfviewer.annotation.core.shapes.Edge
+
+/** Convert image coordinates to pdf coordinates and vice versa, no scale. Y-axis is inverted */
+fun PointF.convertCoordinatesFrom(pageHeight: Int) = PointF(x, pageHeight - y)
+
+/** From the given diagonal bottom left and top right points, calculate the other 2 points */
+fun generateRectangleCoordinates(bottomLeft: PointF, topRight: PointF): List<PointF> {
+    val topLeft = PointF(bottomLeft.x, topRight.y)
+    val bottomRight = PointF(topRight.x, bottomLeft.y)
+
+    return listOf(topLeft, topRight, bottomRight, bottomLeft)
+}
```

Comment:
This rectangle is not the shape rectangle but the rectangle that is around something, for circle we would need it too.  https://github.com/LibrePDF/OpenPDF/blob/master/openpdf/src/main/java/com/lowagie/text/Rectangle.java

I can rename it to `Rect` to have it different from the shape name

Reply:
Ah okay, since its confusing due to the fact that we also have a shape called Rectangle, rename this to `Rect` and also add some documentation stating what ou said in your comment above to avoid future confusion.

Reply:
Improved name and Kdoc